lib/aescrypt.rb

```diff
@@ -77,7 +82,14 @@ def self.encrypt_data(data, key, iv, cipher_type)
     aes = OpenSSL::Cipher::Cipher.new(cipher_type)
     aes.encrypt
     aes.key = key
-    aes.iv = iv if iv != nil
-    aes.update(data) + aes.final      
+
+    iv = aes.random_iv if iv.nil? && cipher_type == 'AES-256-CBC'
+
+    if iv.nil?
+      aes.update(data) + aes.final
```

Comment:
```ruby 
if iv.nil?
```
I don't think it is necessary. It will never be nil. You must check for the length of IV.

```
iv = aes.random_vi if iv.length < 12 && cipher_type.downcase =  'aes-256-cbc'
```